detectable/src/main/java/com/synopsys/integration/detectable/detectables/npm/packagejson/PackageJsonExtractor.java

```diff
@@ -70,7 +70,7 @@ public Extraction extract(CombinedPackageJson combinedPackageJson) {
 
     private List<Dependency> transformDependencies(MultiValuedMap<String, String> dependencies) {
         if (dependencies == null || dependencies.size() == 0) {
-            return Collections.emptyList();
+            return new ArrayList<>();
```

Comment:
The change looks good; however, I am not sure how a "null pointer" was happening before.
Could you share a stack trace or something similar that would shed some light on that?

Reply:
yeah I didn't anticipate one either based on the dependencies checks above this line but I guess we are only protected if this list was never used. I don't have a stack handy but what seemed to be happening was that we entered this block and returned the emptyList. This list can't be modified so it was kind of the wrong thing to do. Later in the code we discover other dependencies and try to add them to this list and it blows up.